Tests/Common/SymbolTests.cs

```diff
@@ -556,6 +557,82 @@ public void TestIfFridayLastTradingDayIsHolidaysThenMoveToPreviousThursday()
             Assert.AreEqual(thursdayBeforeGoodFriday, OptionSymbol.GetLastDayOfTrading(symbol));
         }
 
+        [Test]
+        public void NegativeStrikePriceRoundTrip()
+        {
+            var future = Symbol.CreateFuture(
+                "CL",
+                Market.NYMEX,
+                new DateTime(2020, 5, 20));
+
+            var option = Symbol.CreateOption(
+                future,
+                Market.NYMEX,
+                OptionStyle.American,
+                OptionRight.Call,
+                -50,
+                new DateTime(2020, 4, 16));
+
+            Assert.AreEqual(-50, option.ID.StrikePrice);
+
+            var properties = (ulong)typeof(SecurityIdentifier)
+                .GetField("_properties", BindingFlags.NonPublic | BindingFlags.Instance)
+                .GetValue(option.ID);
+
+            // Forces the reconstruction of the strike price to ensure that it's been properly parsed.
+            var newSid = new SecurityIdentifier(option.ID.Symbol, properties);
+
+            Assert.AreEqual(-50, newSid.StrikePrice);
+        }
+
+        [Test]
+        public void SymbolHashForOptionsBackwardsCompatibilityWholeNumber()
+        {
+            var equity = Symbol.Create("AAPL", SecurityType.Equity, Market.USA);
+            var option = Symbol.CreateOption(
+                equity,
+                Market.USA,
+                OptionStyle.American,
+                OptionRight.Call,
+                100m,
+                new DateTime(2020, 5, 21));
+
+            Assert.AreEqual("AAPL XEOLB4YAQ8BQ|AAPL R735QTJ8XC9X", option.ID.ToString());
+            Assert.AreEqual(100m, option.ID.StrikePrice);
+        }
+
+        [Test]
+        public void SymbolHashForOptionsBackwardsCompatibilityFractionalNumber()
+        {
+            var equity = Symbol.Create("AAPL", SecurityType.Equity, Market.USA);
+            var option = Symbol.CreateOption(
+                equity,
+                Market.USA,
+                OptionStyle.American,
+                OptionRight.Call,
+                0.01m,
+                new DateTime(2020, 5, 21));
+
+            Assert.AreEqual("AAPL XEOLB4YAHNOM|AAPL R735QTJ8XC9X", option.ID.ToString());
+            Assert.AreEqual(0.01m, option.ID.StrikePrice);
+        }
+
+        [Test]
+        public void SymbolHashForOptionsBackwardsCompatibilityLargeFractionalNumberDoesNotThrow()
+        {
+            Assert.DoesNotThrow(() =>
+            {
+                var equity = Symbol.Create("AAPL", SecurityType.Equity, Market.USA);
+                var option = Symbol.CreateOption(
+                    equity,
+                    Market.USA,
+                    OptionStyle.American,
+                    OptionRight.Call,
+                    3600.75m,
+                    new DateTime(2020, 5, 21));
+            });
+        }
+
```

Comment:
Would be nice to add test cases where the strike is off the limits and we assert it behaves as we expect and throws and exception